src/views/common/Animation/AnimationStatePlaybackBarLayer.tsx

```diff
@@ -1,175 +1,156 @@
-        if (norm([x - drawData.seenX, barAreaVCenter - y], 2) < scrubberRadius)
-        {
-            // start dragging
-            draggingXRef.current = x
-            // and pause while dragging, if we weren't already playing
-            if (isPlaying) {
-                wasPlayingRef.current = true
-                dispatch({type: 'TOGGLE_PLAYBACK'})
-            }
-        }
-        else {
-            const newFrame = getNewFrame(x, y)
-            if (!newFrame) return
-            dispatch({type: 'SET_CURRENT_FRAME', newIndex: newFrame})
-        }
-    }, [getEventPoint, drawData.seenX, barAreaVCenter, isPlaying, dispatch, getNewFrame])
-
+        if (initiateScrubbing(x, y, isPlaying)) return // if the user grabbed the scrubber, don't do anything else
+        dragSelectInitiationRef.current = window.setTimeout(() => {
+            dragSelectInitiationRef.current = undefined
+            initiateDragSelection(x)
+        }, dragSelectInitiationDelayMs)
+    }, [getEventPoint, isPlaying, initiateScrubbing, initiateDragSelection])
+    
     const handleMouseUp = useCallback((e: React.MouseEvent) => {
-        if (draggingXRef.current && wasPlayingRef.current && !isPlaying) {
-            dispatch({type: 'TOGGLE_PLAYBACK'})
-            wasPlayingRef.current = false
+        if (dragSelectInitiationRef.current !== undefined) {
+            // mouseup within dragSelectInitiationDelayMs after mousedown === timedout callback hasn't yet fired.
+            // This corresponds to an actual click, not a drag initiation, so do the click handling stuff
+            clearTimeout(dragSelectInitiationRef.current)
+            dragSelectInitiationRef.current = undefined
+            const [x, y] = getEventPoint(e)
+            handleScrollbarClick(x, y)
+            updateSelectedWindow(undefined) // TODO: Check for possible race condition or double-rerender?
+            return
         }
-        draggingXRef.current = undefined
-    }, [dispatch, isPlaying])
-
-    // TODO: Break drag debounce stuff into a more self-contained hook?
-    const dragTargetFrameRef = useRef<number | undefined>(undefined)
-    const nextDebounceRequest = useRef<number | undefined>(undefined)
-    const debounceSetFrame = useCallback(() => {
-        if (draggingXRef.current && dragTargetFrameRef.current) {
-            if (dragTargetFrameRef.current !== currentFrameIndex) {
-                dispatch({type: 'SET_CURRENT_FRAME', newIndex: dragTargetFrameRef.current})
-            }
-            nextDebounceRequest.current = window.requestAnimationFrame(debounceSetFrame)
-        } else {
-            if (nextDebounceRequest.current) {
-                window.cancelAnimationFrame(nextDebounceRequest.current)
-            }
-            nextDebounceRequest.current = undefined
-            dragTargetFrameRef.current = undefined
-        }
-    }, [currentFrameIndex, dispatch])
-
+        terminateScrubbing(isPlaying)
+        terminateDragSelection()
```

Comment:
These guys should be no-ops if the underlying state isn't relevant (i.e. we aren't scrubbing and/or dragging).